spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java

```diff
@@ -270,7 +270,7 @@ public void orphanedFileRemovedWithParallelTasks() throws InterruptedException,
 
     waitUntilAfter(System.currentTimeMillis());
 
-    Set<String> deletedFiles = Sets.newHashSet();
+    Set<String> deletedFiles = ConcurrentHashMap.newKeySet();
```

Comment:
a) Can you please handle in other spark versions too? Flaky test can happen from any spark module.

b) Please also mention that it fixes 
https://github.com/apache/iceberg/issues/8824

Reply:
+1 on fixing this for 3.3 and 3.4